src/app/components/core/north/north-task-modal/north-task-modal.component.ts

```diff
@@ -153,6 +156,7 @@ export class NorthTaskModalComponent implements OnInit, OnChanges {
       this.viewConfigItemComponent.passwordOnChangeFired = false;
       this.viewConfigItemComponent.passwordMatched = true;
     }
+    super.ngOnDestroy(); // call to unsubscribes all subscriptions
```

Comment:
I'm slightly worried, this is calling ngOnDestroy of component. While we are just toggling the modal (part of component). May be there are only modal level variables only today but that is component level life-cycle event.

Reply:
Model is a part of a component but as a child component with separate life cycle. The issue is here the modal ngOnDestory() method not getting called when toggling the modal. So for calling that destroy method we need to  explicitly call the ngOnDestroy method